src/utils/daos_hdlr.c

```diff
@@ -730,118 +778,218 @@ cont_list_attrs_hdlr(struct cmd_args_s *ap)
 
 }
 
-/* cont_get_prop_hdlr() - get container properties */
-int
-cont_get_prop_hdlr(struct cmd_args_s *ap)
+static int
+cont_decode_props(daos_prop_t *props)
 {
-	daos_prop_t		*prop_query;
-	struct daos_prop_entry	*entry;
-	char			type[10] = {};
-	int			rc = 0;
-	uint32_t		i;
-	uint32_t		entry_type;
-
-	/*
-	 * Get all props except the ACL
-	 */
-	prop_query = daos_prop_alloc(DAOS_PROP_CO_NUM - 1);
-	if (prop_query == NULL)
-		return -DER_NOMEM;
-
-	entry_type = DAOS_PROP_CO_MIN + 1;
-	for (i = 0; i < prop_query->dpp_nr; entry_type++) {
-		if (entry_type == DAOS_PROP_CO_ACL)
-			continue; /* skip ACL */
-		prop_query->dpp_entries[i].dpe_type = entry_type;
-		i++;
-	}
 
-	rc = daos_cont_query(ap->cont, NULL, prop_query, NULL);
-	if (rc) {
-		fprintf(stderr, "Container query failed, result: %d\n", rc);
-		D_GOTO(err_out, rc);
-	}
+	struct daos_prop_entry		*entry;
+	char				type[10];
+	int				rc = 0;
 
-	D_PRINT("Container properties :\n");
+	/* unset properties should get default value */
 
-	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_LABEL);
+	entry = daos_prop_entry_get(props, DAOS_PROP_CO_LABEL);
 	if (entry == NULL || entry->dpe_str == NULL) {
 		fprintf(stderr, "label property not found\n");
-		D_GOTO(err_out, rc = -DER_INVAL);
+		rc = -DER_INVAL;
+	} else {
+		D_PRINT("label:\t\t\t%s\n", entry->dpe_str);
 	}
-	D_PRINT("label -> %s\n", entry->dpe_str);
 
-	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_LAYOUT_TYPE);
+	entry = daos_prop_entry_get(props, DAOS_PROP_CO_LAYOUT_TYPE);
 	if (entry == NULL) {
 		fprintf(stderr, "layout type property not found\n");
-		D_GOTO(err_out, rc = -DER_INVAL);
+		rc = -DER_INVAL;
+	} else {
+		daos_unparse_ctype(entry->dpe_val, type);
+		D_PRINT("layout type:\t\t%s ("DF_X64")\n", type,
+			entry->dpe_val);
 	}
-	daos_unparse_ctype(entry->dpe_val, type);
-	D_PRINT("layout type -> "DF_U64"/%s\n", entry->dpe_val, type);
 
-	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_LAYOUT_VER);
+	entry = daos_prop_entry_get(props, DAOS_PROP_CO_LAYOUT_VER);
 	if (entry == NULL) {
 		fprintf(stderr, "layout version property not found\n");
-		D_GOTO(err_out, rc = -DER_INVAL);
+		rc = -DER_INVAL;
+	} else {
+		D_PRINT("layout version:\t\t"DF_U64"\n", entry->dpe_val);
 	}
-	D_PRINT("layout version -> "DF_U64"\n", entry->dpe_val);
 
-	entry = daos_prop_entry_get(prop_query, DAOS_PROP_CO_CSUM);
+	entry = daos_prop_entry_get(props, DAOS_PROP_CO_CSUM);
 	if (entry == NULL) {
 		fprintf(stderr, "checksum type property not found\n");
-		D_GOTO(err_out, rc = -DER_INVAL);
+		rc = -DER_INVAL;
+	} else {
+		D_PRINT("checksum type:\t\t");
+		switch (entry->dpe_val) {
+		case DAOS_PROP_CO_CSUM_OFF:
+			D_PRINT("off\n");
+			break;
+		case DAOS_PROP_CO_CSUM_CRC16:
+			D_PRINT("crc16\n");
+			break;
+		case DAOS_PROP_CO_CSUM_CRC32:
+			D_PRINT("crc32\n");
+			break;
+		case DAOS_PROP_CO_CSUM_CRC64:
+			D_PRINT("crc64\n");
+			break;
+		case DAOS_PROP_CO_CSUM_SHA1:
+			D_PRINT("sha1\n");
+			break;
```

Comment:
We have landed DAOS_PROP_CO_CSUM_SHA256 and DAOS_PROP_CO_CSUM_SHA512 since then. Please provide a follow-up patch adding support for that.